### ISSUE TITLE
asyncFilter, asyncMap, asyncFlatMap with concurrency > 2 emit out of order

### ISSUE THREAD

**Issue by conartist6** [iter-tools/iter-tools#228]

**Comment by sithmel** [comment]
Nice catch!

**Comment by conartist6** [comment]
I was able to fix it and also simplify ParallelRunner quite a bit. I'll put up a PR. These methods are already in the 6.x branch so we should make a patch release I think.

**Comment by sithmel** [comment]
Can we close this ?

**Comment by conartist6** [comment]
Depends on whether or not you want to patch the 6x branch.

**Comment by sithmel** [comment]
Ah ok. I am going to patch and then close

**Comment by conartist6** [comment]
Not patched in 6.x. Fixed in 7.x. I've waffled a lot on whether I planned to try to release a last major update to the 6.x line with many of the fixes in 7.x. I've decided that the answer is no. I'm not doing anything crazy in 7.x. All execution environments supported by 6.x will continue to be. I do make a few breaking API changes, but given that that is the case I think there's no harm in using those fixes to entice people to upgrade.
